### PR TITLE
feat(file): add accept prop to filter selectable file types

### PR DESCRIPTION
**Pull request by sampsonbryce**
fix: #1174

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/file/examples/file-accepted-types.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { FileInfo } from '@limetech/lime-elements';
```
Comment (adrianschmidt) [inline src/components/file/examples/file-accepted-types.tsx:1]: I added `fix: #1174` to the commit message and the PR description. That way, when the fix is released, the changelog will contain links to the commit and the issue, like [this](https://github.com/Lundalogik/lime-elements/releases/tag/v30.15.2).

Comment (adrianschmidt) [inline src/components/file/examples/file-accepted-types.tsx:1]: I added `fix: #1174` to the commit message and the PR description. That way, when the fix is released, the changelog will contain links to the commit and the issue, like [this](https://github.com/Lundalogik/lime-elements/releases/tag/v30.15.2).

Comment (sampsonbryce) [reply src/components/file/examples/file-accepted-types.tsx:1]: Thanks Adrian!

File: src/components/file/file.tsx [hunk 2]
```diff
@@ -80,6 +81,14 @@ export class File {
     @Prop({ reflect: true })
     public disabled: boolean = false;
 
+    /**
+     * The accepted file types
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers
```
Comment (BregenzerK) [inline src/components/file/file.tsx:87]: Really nice and necessary to have this resource in the docs string. I noticed that I couldn't click it in the documentation. could that work if you replace `@see` with `@link`?

**Review by BregenzerK — COMMENTED** [review]
I would approve the PR besides that one remark. @adrianschmidt is it ok if we merge it?

**Comment by adrianschmidt** [comment]
> I would approve the PR besides that one remark. @adrianschmidt is it ok if we merge it?

Yeah, looks good to me! If `@link` doesn't work, maybe you can just add the link with the regular markdown syntax?

**Comment by sampsonbryce** [comment]
@BregenzerK I changed to just a markdown link

**Review by BregenzerK — APPROVED** [review]

**Comment by adrianschmidt** [comment]
Aside:

> > I would approve the PR besides that one remark. @adrianschmidt is it ok if we merge it?
> 
> Yeah, looks good to me! If `@link` doesn't work, maybe you can just add the link with the regular markdown syntax?

Ah, come to think of it, I think `@link` isn't for adding a url. I _think_ it's for adding a link to another component in the documentation.

Regardless, the markdown link works just fine 👌

File: src/components/file/file.tsx [hunk 3]
```diff
@@ -80,6 +81,12 @@ export class File {
     @Prop({ reflect: true })
     public disabled: boolean = false;
 
+    /**
+     * The [accepted file types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers)
+     */
+    @Prop({ reflect: true })
+    public accept: string = '*';
```
Comment (adrianschmidt) [inline src/components/file/file.tsx:88]: (This isn't an issue I think should prevent merging this, just a small docs thing Johan and I might want to look into.)

Hmm… despite a default value being set here, it doesn't show up in the docs:

image

@jgroth Do you know off the top of your head what the reason might be? I've looked through some of our other components, but we don't have many string properties with default values. We do have `type` on limel-slider, which defaults to `''`, and that shows up as "Default value: `false`" in the docs… so I think the value is being handled incorrectly somewhere. I'm not sure if it's in Kompendium or in some lib we're using though…

image

image

**Pull request merged**